pkg/resources/row_access_policy_acceptance_test.go

```diff
@@ -32,54 +42,393 @@ func TestAcc_RowAccessPolicy(t *testing.T) {
 			{
 				ConfigDirectory: config.TestStepDirectory(),
 				ConfigVariables: m(),
-				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "name", id.Name()),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "fully_qualified_name", id.FullyQualifiedName()),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "database", acc.TestDatabaseName),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "schema", acc.TestSchemaName),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "comment", "Terraform acceptance test"),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "row_access_expression", "case when current_role() in ('ANALYST') then true else false end"),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "signature.N", "VARCHAR"),
-					resource.TestCheckResourceAttr("snowflake_row_access_policy.test", "signature.V", "VARCHAR"),
+				Check: assert.AssertThat(t, resourceassert.RowAccessPolicyResource(t, resourceName).
+					HasNameString(id.Name()).
+					HasDatabaseString(id.DatabaseName()).
+					HasSchemaString(id.SchemaName()).
+					HasFullyQualifiedNameString(id.FullyQualifiedName()).
+					HasCommentString("Terraform acceptance test").
+					HasBodyString("case when current_role() in ('ANALYST') then true else false end"),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.#", "2")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.0.name", "N")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.0.type", "VARCHAR")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.1.name", "V")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "argument.1.type", "VARCHAR")),
+					resourceshowoutputassert.RowAccessPolicyShowOutput(t, resourceName).
+						HasCreatedOnNotEmpty().
+						HasDatabaseName(id.DatabaseName()).
+						HasKind(string(sdk.PolicyKindRowAccessPolicy)).
+						HasName(id.Name()).
+						HasOptions("").
+						HasOwner(snowflakeroles.Accountadmin.Name()).
+						HasOwnerRoleType("ROLE").
+						HasSchemaName(id.SchemaName()).
+						HasComment("Terraform acceptance test"),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.body", "case when current_role() in ('ANALYST') then true else false end")),
```

Comment:
I think there was example in user on creating a simple ext to assert show output. I guess it could be easily altered to make it work with desc output.

Reply:
it's a bit different in user because what was added was to assert showoutput inside a datasource. But. There is another option of adding the describe to the resourceshowoutputassert by adding it to the `objectassertgen.GetSdkObjectDetails`, generating, and adding a proper constructors as ext. But I don't know if this is worth a hassle now.

Reply:
I left this because I assume we'd like to have it in a separate pkg, like `resourcedescribeoutputassert`.